apps/condo/domains/notification/tasks/deliverMessage.js

```diff
@@ -46,132 +50,124 @@ async function _sendMessageByAdapter (transport, adapter, messageContext) {
             const messageContext = await adapter.prepareMessageToSend(message)
+
             processingMeta.messageContext = messageContext
+            transportMeta.messageContext = messageContext
+            processingMeta.transports.push(transport)
 
             const [isOk, deliveryMetadata] = await _sendMessageByAdapter(transport, adapter, messageContext)
+
             processingMeta.deliveryMetadata = deliveryMetadata
             transportMeta.deliveryMetadata = deliveryMetadata
+            transportMeta.status = isOk ? MESSAGE_SENT_STATUS : MESSAGE_ERROR_STATUS
+            processingMeta.step = isOk ? MESSAGE_SENT_STATUS : MESSAGE_ERROR_STATUS
+            successCnt += isOk ? 1 : 0
 
-            if (isOk) {
-                transportMeta.status = MESSAGE_SENT_STATUS
-                processingMeta.step = MESSAGE_SENT_STATUS
-                break
-            } else {
-                transportMeta.status = MESSAGE_ERROR_STATUS
-                processingMeta.step = MESSAGE_ERROR_STATUS
-            }
         } catch (error) {
             transportMeta.status = MESSAGE_ERROR_STATUS
             transportMeta.exception = safeFormatError(error, false)
-            logger.error({ msg: 'deliverMessage error', error, messageId, transportMeta, transportsMeta, processingMeta })
+
+            logger.error({ msg: 'deliverMessage error', error, messageId, transportMeta, processingMeta })
         }
+
+        processingMeta.transportsMeta.push(transportMeta)
+
+        if (sendByOneTransport && successCnt > 0) break
     }
 
-    // message sent either directly or by fallback transport
-    const status = (processingMeta.step === MESSAGE_SENT_STATUS) ? MESSAGE_SENT_STATUS : MESSAGE_ERROR_STATUS
-    await Message.update(keystone, message.id, {
+    processingMeta.status = isEmpty(processingMeta.transportsMeta)
+        ? '-'
+        :  processingMeta.transportsMeta.map(x => `${x.transport}:${x.status}`).join(',')
+
+    const messageFinalData = {
         ...baseAttrs,
-        status,
-        sentAt: (processingMeta.step === MESSAGE_SENT_STATUS) ? new Date().toISOString() : null,
+        status: (successCnt > 0) ? MESSAGE_SENT_STATUS : MESSAGE_ERROR_STATUS,
+        sentAt: (successCnt > 0) ? new Date().toISOString() : null,
         deliveredAt: null,
         readAt: null,
         processingMeta,
-    })
+    }
+
+    await Message.update(context, message.id, messageFinalData)
 
-    return status
+    return messageFinalData.status
 }
 
+const deliverMessageFast = async (messageId) => await deliverMessage(messageId)
+const deliverMessageDefault = async (messageId) => await deliverMessage(messageId)
+const deliverMessageSlow = async (messageId) => await deliverMessage(messageId)
+
 module.exports = {
-    deliverMessage: createTask('deliverMessage', deliverMessage, { priority: 1 }),
+    deliverMessageFast: createTask('deliverMessageFast', deliverMessageFast, { priority: 1 }),
```

Comment:
It may confuse other developers. Probably, you should export only one task and that task should decide bout the priority. Doesn't look like bullet proven solution. We need to think about this